### PR TITLE
Site-specific logout

### PR DESCRIPTION
**Pull request by zamzterz**
This depends on new site-specific logout endpoint in session-service, so don't merge yet 🙂

### CODE REVIEW

**Comment by zamzterz** [comment]
@marjuszkiewicz The new endpoint has been added in session-service, so this is now ready for review and testing.

File: src/identity.js [hunk 1]
```diff
@@ -782,10 +769,12 @@ export class Identity extends EventEmitter {
      */
     logoutUrl(redirectUri = this.redirectUri) {
         assert(isUrl(redirectUri), `logoutUrl(): redirectUri is invalid`);
-        return this._spid.makeUrl('logout', {
-            response_type: 'code',
-            redirect_uri: redirectUri
-        });
+        const params = { redirect_uri: redirectUri };
+        if (this._sessionService) {
```
Comment (marjuszkiewicz) [inline src/identity.js]: This seems to be breaking change, how about adding new option to constructor https://github.com/schibsted/account-sdk-browser/blob/master/src/identity.js#L90 ? `singleSiteLogin` with default set to `false` maybe?

File: src/identity.js [hunk 2]
```diff
@@ -782,10 +769,12 @@ export class Identity extends EventEmitter {
...
+            return this._sessionService.makeUrl('logout', params);
+        } else {
```
Comment (marjuszkiewicz) [inline src/identity.js]: NIT: don't need to have `else` block here, since there is `return` in if statement.

File: src/identity.js [hunk 3]
```diff
@@ -441,20 +441,7 @@ export class Identity extends EventEmitter {
...
-                        // site switched to using the session-service. If the request contains a
-                        // session-cookie but no session is found (return code will be 404), then we
-                        // *should* throw an exception and *not* fall through to spid-hassession
-                        if (err.code !== 400) {
-                            this.emit('error', err);
-                            return reject(new SDKError('HasSession failed', err));
-                        }
-                        data = null;
-                    }
+                    data = await this._sessionService.get('/session');
```
Comment (marjuszkiewicz) [inline src/identity.js]: Since it is also braking change, how about using `singleSiteLogin` config here as well?

**Pull request merged**